src/test/kotlin/org/rustPerformanceTests/refactoring/RsInlineFunctionRandomRefactoringTest.kt

```diff
@@ -0,0 +1,149 @@
+
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import org.rust.cargo.project.settings.toolchain
+import org.rust.cargo.toolchain.tools.CargoCheckArgs
+import org.rust.ide.annotator.RsExternalLinterUtils
+import org.rust.ide.refactoring.inlineFunction.RsInlineFunctionProcessor
+import org.rust.lang.RsFileType
+import org.rust.lang.core.macros.MacroExpansionScope
+import org.rust.lang.core.macros.macroExpansionManager
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ref.RsReference
+import org.rust.openapiext.document
+import org.rust.openapiext.toPsiFile
+import org.rustPerformanceTests.RsRealProjectTestBase
+import org.rustPerformanceTests.findDescendants
+import kotlin.random.Random
+
+private const val ONLY_SAME_MOD: Boolean = true
+private const val ONLY_FUNCTIONS: Boolean = false
+
+/**
+ * For testing [RsInlineFunctionProcessor] on real projects.
+ * Workflow:
+ * - Open some real-world project
+ * - Choose one function among all project files
+ * - Invoke refactoring to inline this function
+ * - Run `cargo check` to check for possible errors after inlining
+ */
+class RsInlineFunctionRandomRefactoringTest : RsRealProjectTestBase() {
+
+    // when testing on cargo, remove `#![cfg_attr(test, deny(warnings))]` in `src/cargo/lib.rs`
+    fun `test Cargo`() = doTest(CARGO)
+    fun `test clap`() = doTest(CLAP)
+    fun `test tokio`() = doTest(TOKIO)
+
+    private fun doTest(info: RealProjectInfo) {
+        project.macroExpansionManager.setUnitTestExpansionModeAndDirectory(MacroExpansionScope.ALL, name)
+        val base = openRealProject(info) ?: error("Can't open project")
+
+        val files = collectFiles(base)
+        invokeCargoCheck(files.first().first)
+
+        val seed = Random.nextInt()
+        println("seed: $seed")
+        val random = Random(seed)
+        for (index in 0..Int.MAX_VALUE) {
+            val (file, functions) = files.random(random)
+            val (function, usage) = chooseRandomFunctionAndUsage(functions, random) ?: continue
+            invokeRefactoring(function, usage, index)
+            invokeCargoCheck(file)
+        }
+    }
+
+    private fun invokeRefactoring(function: RsFunction, usage: RsReference, index: Int) {
+        val call = usage.element
+        if (!call.isMethodOrFunctionCall()) return
+
+        val callText = call.formatMethodOrFunctionCall().substringBefore('\n')
+        val line = call.containingFile.document!!.getLineNumber(call.startOffset) + 1
+        println("#$index Inlining `$callText` in ${call.containingMod.qualifiedName}  :$line")
+        val processor = RsInlineFunctionProcessor(project, function, usage, inlineThisOnly = true, removeDefinition = false)
+        processor.run()
+    }
+
+    private fun collectFiles(base: VirtualFile): List<Pair<RsFile, List<RsFunction>>> =
+        base
+            .findDescendants { it.fileType == RsFileType }
```

Comment:
This logic seems to duplicate `val filesToCheck = baseDirToCheck.findDescendants {...}` from `org.rustPerformanceTests.RsRealProjectAnalysisTest#doTest`. Maybe this could be refactored